plasmapy/dispersion/two_fluid_dispersion_solver.py

```diff
@@ -0,0 +1,243 @@
+
+    Parameters
+    ----------
+    B : ~astropy.units.Quantity
+        Magnetic field
+    gamma_e : float or int
+        The adiabatic index for electrons, which defaults to 1.  This
+        value assumes that the electrons are able to equalize their
+        temperature rapidly enough that the electrons are effectively
+        isothermal.
+    gamma_i : float or int
+        The adiabatic index for ions, which defaults to 3. This value
+        assumes that ion motion has only one degree of freedom, namely
+        along magnetic field lines.
+    ion : string, optional
+        Representation of the ion species (e.g., `'p'` for protons,
+        `'D+'` for deuterium, or 'He-4 +1' for singly ionized
+        helium-4), which defaults to protons.  If no charge state
+        information is provided, then the ions are assumed to be
+        singly charged.
+    k : ~astropy.units.Quantity
+        Wave number
+    m_e : ~astropy.units.Quantity
+        Mass of negative ion which defaults to electron
+    m_i : ~astropy.units.Quantity
+        Mass of positive ion which defaults to proton
+    n : ~astropy.units.Quantity
+        Number density.
+    T_e : ~astropy.units.Quantity
+        The electron temperature
+    T_i : ~astropy.units.Quantity
+        The ion temperature
+    theta : ~astropy.units.Quantity
+        Angle of propagation which defaults to 45 degrees.
+    z : float or integer
+        Average ionization number which defaults to 1 for protons.
+
+    Returns
+    -------
+    omega : ~astropy.units.Quantity
+        A dictionary of Wave frequencies corresponding to three modes namely
+        a) Ion-acoustic mode, b) Alfven mode and c) Fast mode, in 1/s units.
+
+    Raises
+    ------
+    TypeError
+        The magnetic field and density arguments are not instances of
+        `~astropy.units.Quantity` and cannot be converted into those.
+
+    ~astropy.units.UnitConversionError
+        If the magnetic field or density is not in appropriate units.
+
+    ValueError
+        If the magnetic field, density or wavenumber is negative, or the ion
+        mass or charge state cannot be found.
+
+    Notes
+    -----
+    Computes the solution for wave dispersion relation based on equation 38 of
+    [1]
+
+    .. math::
+        \frac{\omega}{\omega_{ci}} = \sqrt(2\Lambda \sqrt(-\frac{p}{3}) cos
+        \left( \frac{1}{3} cos^{-1}\left( \frac{3q}{2p} \sqrt(-\frac{3}{p}
+        \right) - \frac{2\pi}{3}j \right) + \frac{A\Lambda}{3})
+
+    Where,
+        j = 0 ==> fast mode
+        j = 1 ==> Alfven mode
+        j = 2 ==> Acoustic mode
+
+    The above equation is derived from the general wave equation in the low
+    frequency regime, where both electrons and ions play significant role (in
+    the high frequency regime ions do not play any significant role and the
+    process is dominated by electron dynamics).
+
+    The complete dispersion equation is thus written as (from equation (1) of
+    [1]):
```

Comment:
```suggestion
    [1]_):
```

The trailing underscore tells `Sphinx` this is suppose to be a link.